src/ocrd_network/processor_server.py

```diff
@@ -76,74 +66,70 @@ def __init__(self, mongodb_addr: str, processor_name: str = "", processor_class=
             raise Exception(f"The ocrd_tool is empty or missing")
 
         if not self.processor_name:
-            self.processor_name = self.ocrd_tool['executable']
+            self.processor_name = self.ocrd_tool["executable"]
+
+    async def on_startup(self):
+        await initiate_database(db_url=self.db_url)
+
+    async def on_shutdown(self) -> None:
+        """
+        TODO: Perform graceful shutdown operations here
+        """
+        pass
 
-        # Create routes
-        self.router.add_api_route(
-            path='/info',
+    def add_api_routes_processing(self):
```

Comment:
I think a call of this method probably in the constructor is missing. Or anywhere else, it is not used anywhere.

Reply:
Good catch. There are not any processor server tests yet so I missed that.